src/js/actions/export.js

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
+ *  
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"), 
+ * to deal in the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+define(function (require, exports) {
+    "use strict";
+
+    var Promise = require("bluebird"),
+        Immutable = require("immutable");
+
+    var dialog = require("./dialog"),
+        events = require("js/events"),
+        locks = require("js/locks"),
+        collection = require("js/util/collection"),
+        log = require("js/util/log"),
+        ExportAsset = require("js/models/exportasset"),
+        ExportService = require("js/util/exportservice");
+
+    var descriptor = require("adapter/ps/descriptor"),
+        // documentLib = require("adapter/lib/document"),
+        layerLib = require("adapter/lib/layer");
+
+    /**
+     * Export metadata is stored in the "extension" data structure in Ps, using this as the namespace
+     * @type {String}
+     */
+    var EXTENSION_DATA_NAMESPACE = "designSpace";
+
+    /**
+     * An instance of the ExportService utility used to communicate with the generator plugin
+     * @private
+     * @type {ExportService}
+     */
+    var _exportService;
+
+    /**
+     * before start up, initialize the export service
+     * @return {Promise}
+     */
+    var beforeStartup = function () {
```

Comment:
All the lifecycle functions (`beforeStartup`, etc.) should go at the bottom of the file.